builtin/blame.c

```diff
@@ -319,18 +319,20 @@ static const char *format_time(timestamp_t time, const char *tz_str,
 	return time_buf.buf;
 }
 
-#define OUTPUT_ANNOTATE_COMPAT	001
-#define OUTPUT_LONG_OBJECT_NAME	002
-#define OUTPUT_RAW_TIMESTAMP	004
-#define OUTPUT_PORCELAIN	010
-#define OUTPUT_SHOW_NAME	020
-#define OUTPUT_SHOW_NUMBER	040
-#define OUTPUT_SHOW_SCORE	0100
-#define OUTPUT_NO_AUTHOR	0200
-#define OUTPUT_SHOW_EMAIL	0400
-#define OUTPUT_LINE_PORCELAIN	01000
-#define OUTPUT_COLOR_LINE	02000
-#define OUTPUT_SHOW_AGE_WITH_COLOR	04000
+enum output_type {
+	OUTPUT_ANNOTATE_COMPAT = 001,
+	OUTPUT_LONG_OBJECT_NAME = 002,
+	OUTPUT_RAW_TIMESTAMP = 004,
+	OUTPUT_PORCELAIN = 010,
+	OUTPUT_SHOW_NAME = 020,
+	OUTPUT_SHOW_NUMBER = 040,
+	OUTPUT_SHOW_SCORE = 0100,
+	OUTPUT_NO_AUTHOR = 0200,
+	OUTPUT_SHOW_EMAIL = 0400,
+	OUTPUT_LINE_PORCELAIN = 01000,
+	OUTPUT_COLOR_LINE = 02000,
+	OUTPUT_SHOW_AGE_WITH_COLOR = 04000
+};
```

Comment:
This makes sense to me, but please note that these are bit field constants, and it seems that the jury is still out there whether it is a good idea to turn those into `enum`s.

See https://public-inbox.org/git/xmqqk19fn9jp.fsf@gitster-ct.c.googlers.com/ and the surrounding thread.

If you want to take this to the Git mailing list, I would suggest to align the constants, and to turn them into `1<<0`, `1<<1`, `1<<2` etc.

Reply:
As this pull request is about "Turning Preprocessor constants into enum".
Should I target the different file to turn preprocessor constant into enum in this PR?
or 
turn these ones into form of `1<<0`, `1<<1` etc

Reply:
I strongly suggest focusing on one case at a time, i.e. align the values and turn them into `1<<0`, `1<<1`, etc

Reply:
(And adjusting the header of the commit message to talk about this concrete instance instead of suggesting that the patch takes care of _all_ preprocessor constants that could potentially be turned into `enum`s.)

Reply:
> I strongly suggest focusing on one case at a time, i.e. align the values and turn them into `1<<0`, `1<<1`, etc

> (And adjusting the header of the commit message to talk about this concrete instance instead of suggesting that the patch takes care of _all_ preprocessor constants that could potentially be turned into `enum`s.)

Ok. Looking forward to this